### PR TITLE
FW: Clean the cmdlets code, add a deprecation warning to some of the …

### PR DESCRIPTION
**Pull request by gucalder**
…cmdlets, add some verbose messages

### CODE REVIEW

**Comment by hovsepm** [comment]
@gucalder please pull from upstream dev and resolve conflicts

**Comment by gucalder** [comment]
I see error when compiling Compute.sln (locally) and a missing file in the build above.

**Comment by markcowl** [comment]
@gucalder you are failing now due to msi build issues.  I'm not sure why your PR would involve the msi changes you have - Storage and the associated files are required for insights, and your PR should have no bearing on the storage solution.  You can either remove the wix changes, or do a truly clean build (```git clean -xdf``` after committing your changes)  before regenerating the wix file.

**Comment by gucalder** [comment]
Done, I removed the wxi changes and that resolved the issues.

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/561/

File: src/ResourceManager/Insights/Commands.Insights/Alerts/AddAlertRuleCommandBase.cs [hunk 1]
```diff
@@ -116,7 +116,10 @@ public abstract class AddAlertRuleCommandBase : ManagementCmdletBase
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Alerts/AddAlertRuleCommandBase.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Alerts/GetAlertHistoryCommand.cs [hunk 2]
```diff
@@ -71,6 +71,8 @@ protected override bool KeepTheRecord(EventData record)
         /// <returns>Modified query filter including the condition for eventSource</returns>
         protected override string ProcessParticularParameters(string currentQueryFilter)
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Alerts/GetAlertHistoryCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Alerts/GetAlertRuleCommand.cs [hunk 3]
```diff
@@ -71,6 +71,8 @@ public class GetAlertRuleCommand : ManagementCmdletBase
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Alerts/GetAlertRuleCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Alerts/RemoveAlertRuleCommand.cs [hunk 4]
```diff
@@ -50,6 +49,8 @@ public class RemoveAlertRuleCommand : ManagementCmdletBase
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Alerts/RemoveAlertRuleCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/AddAutoscaleSettingCommand.cs [hunk 5]
```diff
@@ -94,7 +93,10 @@ public class AddAutoscaleSettingCommand : ManagementCmdletBase
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/AddAutoscaleSettingCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/GetAutoscaleHistoryCommand.cs [hunk 6]
```diff
@@ -71,6 +71,8 @@ protected override bool KeepTheRecord(EventData record)
         /// <returns>Modified query filter including the condition for eventSource</returns>
         protected override string ProcessParticularParameters(string currentQueryFilter)
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/GetAutoscaleHistoryCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/GetAutoscaleSettingCommand.cs [hunk 7]
```diff
@@ -59,6 +58,8 @@ public class GetAutoscaleSettingCommand : ManagementCmdletBase
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/GetAutoscaleSettingCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/NewAutoscaleProfileCommand.cs [hunk 8]
```diff
@@ -136,6 +136,8 @@ public class NewAutoscaleProfileCommand : AzureRMCmdlet
         /// </summary>
         public override void ExecuteCmdlet()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/NewAutoscaleProfileCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/NewAutoscaleRuleCommand.cs [hunk 9]
```diff
@@ -116,6 +113,8 @@ public class NewAutoscaleRuleCommand : AzureRMCmdlet
         /// </summary>
         public override void ExecuteCmdlet()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/NewAutoscaleRuleCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/RemoveAutoscaleSettingCommand.cs [hunk 10]
```diff
@@ -50,6 +48,8 @@ public class RemoveAutoscaleSettingCommand : ManagementCmdletBase
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/RemoveAutoscaleSettingCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/ManagementCmdletBase.cs [hunk 11]
```diff
@@ -35,11 +35,10 @@ public abstract class ManagementCmdletBase : InsightsCmdletBase, IDisposable
         /// </summary>
         public IInsightsManagementClient InsightsManagementClient
         {
-            get
-            {
+            // The premise is that a command to establish a context (like Add-AzureRmAccount) has been called before this command in order to have a correct CurrentContext
+            get {
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/ManagementCmdletBase.cs]: Please make line break in the comment at around 100th character. It is extremely hard to read long comments on github for anyone.

File: src/ResourceManager/Insights/Commands.Insights/Metrics/FormatMetricsAsTableCommand.cs [hunk 12]
```diff
@@ -77,6 +76,8 @@ public PSMetricTabularResult[] ProcessParameter()
         /// </summary>
         public override void ExecuteCmdlet()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Metrics/FormatMetricsAsTableCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Metrics/GetMetricDefinitionsCommand.cs [hunk 13]
```diff
@@ -71,12 +70,15 @@ protected string ProcessParameters()
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Metrics/GetMetricDefinitionsCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/Metrics/GetMetricsCommand.cs [hunk 14]
```diff
@@ -127,12 +127,15 @@ protected string ProcessParameters()
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/Metrics/GetMetricsCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Commands.Insights/UsageMetrics/GetUsageMetricsCommand.cs [hunk 15]
```diff
@@ -121,6 +121,8 @@ protected string ProcessParameters()
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            WriteWarning("This cmdlet is being modified to enable better experience and may contain breaking changes in a future release.\n\r");
+
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights/UsageMetrics/GetUsageMetricsCommand.cs]: Please, remove \n\r in the end.

File: src/ResourceManager/Insights/Insights.sln [hunk 16]
```diff
@@ -12,8 +12,6 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Insights.Test", "C
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.ScenarioTests.ResourceManager.Common", "..\Common\Commands.ScenarioTests.ResourceManager.Common\Commands.ScenarioTests.ResourceManager.Common.csproj", "{3436A126-EDC9-4060-8952-9A1BE34CDD95}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Profile", "..\Profile\Commands.Profile\Commands.Profile.csproj", "{142D7B0B-388A-4CEB-A228-7F6D423C5C2E}"
-EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Common", "..\..\Common\Commands.Common\Commands.Common.csproj", "{5EE72C53-1720-4309-B54B-5FB79703195F}"
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Insights.sln]: Why did you remove Profile? so if anyone would like to debug your cmdlet they will have to build all the repo and reference AzureResourceManager.psd1. This project is in the solution just for development/issue debugging. please revert this change.

File: src/ResourceManager/Insights/Commands.Insights.Test/packages.config [hunk 17]
```diff
@@ -24,7 +24,6 @@
   <package id="System.Spatial" version="5.6.4" targetFramework="net45" />
   <package id="WindowsAzure.Storage" version="6.0.0" targetFramework="net45" />
   <package id="xunit" version="2.1.0" targetFramework="net45" />
-  <package id="xunit.abstractions" version="2.0.0" targetFramework="net45" />
   <package id="xunit.assert" version="2.1.0" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Insights/Commands.Insights.Test/packages.config]: revert this change.

**Comment by hovsepm** [comment]
please re-target your changes to release-1.2.2 branch.
